android/hello_sdl_android/src/main/java/com/sdl/hellosdlandroid/SdlReceiver.java

```diff
@@ -56,4 +76,43 @@ public void onReceive(Context context, Intent intent) {
     public String getSdlServiceName() {
         return SdlService.class.getSimpleName();
     }
+
+    private final BroadcastReceiver mUsbReceiver = new BroadcastReceiver() {
+
+        public void onReceive(Context context, Intent intent) {
+            String action = intent.getAction();
+            if (ACTION_USB_PERMISSION.equals(action) && context != null && cachedIntent != null && pendingIntent != null) {
+                if (AndroidTools.hasForegroundServiceTypePermission(context)) {
+                    try {
+                        pendingIntent.send(context, 0, cachedIntent);
+                    } catch (PendingIntent.CanceledException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }
+    };
+
+    /**
+     * Request permission from USB Accessory if USB accessory is not null.
+     * If the user has not granted the BLUETOOTH_CONNECT permission,
+     * we can request the USB Accessory permission to satisfy the requirements for
+     * FOREGROUND_SERVICE_CONNECTED_DEVICE and can start our service and allow
+     * it to enter the foreground. FOREGROUND_SERVICE_CONNECTED_DEVICE is a requirement
+     * in Android 14
+     */
+    private void requestUsbAccessory(Context context) {
+        UsbManager manager = (UsbManager) context.getSystemService(Context.USB_SERVICE);
+        if (manager.getAccessoryList() == null) {
+            return;
+        }
+        PendingIntent mPermissionIntent = PendingIntent.getBroadcast(context, 0, new Intent(ACTION_USB_PERMISSION), PendingIntent.FLAG_IMMUTABLE);
+        IntentFilter filter = new IntentFilter(ACTION_USB_PERMISSION);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+            context.registerReceiver(mUsbReceiver, filter, Context.RECEIVER_EXPORTED);
+        }
```

Comment:
The new method that was added to the utility class can be used here. 

```suggestion
       AndroidTools.registerReceiver(context, mUsbReceiver, filter, Context.RECEIVER_EXPORTED);
```